pkg/cluster/docker_for_mac.go

```diff
@@ -0,0 +1,199 @@
+package cluster
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"path/filepath"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
+	klog "k8s.io/klog/v2"
+)
+
+type d4mSettings map[string]interface{}
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// Uses the DockerForMac GUI protocol to control DockerForMac.
+type DockerForMacClient struct {
+	httpClient HTTPClient
+	socketPath string
+}
+
+func NewDockerForMacClient() (DockerForMacClient, error) {
+	homedir, err := homedir.Dir()
+	if err != nil {
+		return DockerForMacClient{}, err
+	}
+
+	socketPath := filepath.Join(homedir, "Library/Containers/com.docker.docker/Data/gui-api.sock")
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(ctx context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", socketPath)
+			},
+		},
+	}
+	return DockerForMacClient{
+		httpClient: httpClient,
+		socketPath: socketPath,
+	}, nil
+}
+
+func (c DockerForMacClient) writeSettings(ctx context.Context, settings map[string]interface{}) error {
+	klog.V(7).Infof("POST %s /settings\n", c.socketPath)
+	buf := bytes.NewBuffer(nil)
+	err := json.NewEncoder(buf).Encode(c.settingsForWrite(settings))
+	if err != nil {
+		return errors.Wrap(err, "writing d4m settings")
+	}
+
+	klog.V(8).Infof("Request body: %s\n", buf.String())
+	req, err := http.NewRequest("POST", "http://localhost/settings", buf)
```

Comment:
whoa wait, when did this become a valid endpoint? 🤯

Reply:
the HTTP 1.1 protocol needs a HOST, but since we're dialing over a socket it doesn't really matter what it is.

Reply:
ah gotcha. worth commenting imo (maybe make the url a constant and comment there?)